File: solutions/1528/js/solution.js
// 1528. Shuffle String
var restoreString = function(s, indices)
{
	let characterArray = new Array(s.length);

	for (let i = 0; i < s.length; ++i)
	{
		characterArray[indices[i]] = s[i];
	}

	return characterArray.join("");
};

File: solutions/1528/c++/solution.cpp
// 1528. Shuffle String
#include <string>
#include <vector>

using namespace std;

class Solution
{
	public:
	string restoreString(string s, vector<int>& indices)
	{
		string result(s.size(), ' ');

		for (int i = 0; i < s.size(); ++i)
		{
			result[indices[i]] = s[i];
		}

		return result;
	}
};